PocketKit/Sources/PocketKit/Tags/TagsFilterViewModel.swift

```diff
@@ -50,22 +49,25 @@ class TagsFilterViewModel: ObservableObject {
     }
 
     func delete(tags: [String]) {
-        let event = SnowplowEngagement(type: .general, value: nil)
-        let contexts: Context = UIContext.button(identifier: .tagsDelete)
-        tracker.track(event: event, [contexts])
+        trackTagsDelete()
         tags.forEach { tag in
             guard let tag: Tag = fetchedTags.filter({ $0.name == tag }).first else { return }
             source.deleteTag(tag: tag)
         }
     }
 
-    func rename(from oldName: String, to newName: String) {
-        let event = SnowplowEngagement(type: .general, value: nil)
-        let contexts: Context = UIContext.button(identifier: .tagsSaveChanges)
-        tracker.track(event: event, [contexts])
-        guard let tag: Tag = fetchedTags.filter({ $0.name == oldName }).first else { return }
+    func rename(from oldName: String?, to newName: String, completion: () -> Void) {
+        guard let oldName else {
+            Log.capture(message: "Unable to rename tag")
+            completion()
+            return
+        }
+        let newName = newName.lowercased()
+        guard let tag: Tag = fetchedTags.filter({ $0.name == oldName }).first,
+              !fetchedTags.compactMap({ $0.name }).contains(newName) else { return }
```

Comment:
oops forgot one more completion and should prob log this too

Reply:
will most likely change with this ticket https://getpocket.atlassian.net/browse/IN-1350